File: remap_user_references.py
import re
import os

# Set the number of users currently in local_users.dart
USER_COUNT = 21  # Update this to match your actual user count

# List of files to process
FILES = [
    'lib/data/dummy_projects.dart',
    'lib/data/seed_events.dart',
    'lib/data/dummy_marketplace.dart',
]

# Patterns to match localUsers[XX] and 'uXX'
INDEX_PATTERN = re.compile(r'localUsers\[(\d+)\]')
ID_PATTERN = re.compile(r"'u(\d+)'")


def remap_index(match):
    idx = int(match.group(1))
    new_idx = idx % USER_COUNT
    return f'localUsers[{new_idx}]'


def remap_id(match):
    idx = int(match.group(1))
    new_idx = ((idx - 1) % USER_COUNT) + 1
    return f"'u{new_idx}'"


def process_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    content = INDEX_PATTERN.sub(remap_index, content)
    content = ID_PATTERN.sub(remap_id, content)
    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(content)
    print(f'Remapped user references in {filepath}')


def main():
    for file in FILES:
        if os.path.exists(file):
            process_file(file)
        else:
            print(f'File not found: {file}')

if __name__ == '__main__':
    main() 

File: lib/utils/remap_user_references.dart
import 'dart:io';

// Set the number of users currently in local_users.dart
const int userCount = 21; // Update this to match your actual user count

final files = [
  'lib/data/seed_events.dart',
  'lib/data/dummy_marketplace.dart',
];

final indexRegExp = RegExp(r'localUsers\[(\d+)\]');
final idRegExp = RegExp(r"'u(\d+)'", multiLine: true);

String remapIndices(String content) {
  return content.replaceAllMapped(indexRegExp, (match) {
    final idx = int.parse(match.group(1)!);
    final newIdx = idx % userCount;
    return 'localUsers[[0m$newIdx]';
  });
}

String remapIds(String content) {
  return content.replaceAllMapped(idRegExp, (match) {
    final idx = int.parse(match.group(1)!);
    final newIdx = ((idx - 1) % userCount) + 1;
    return "'u$newIdx'";
  });
}

void processFile(String path) {
  final file = File(path);
  if (!file.existsSync()) {
    print('File not found: $path');
    return;
  }
  String content = file.readAsStringSync();
  content = remapIndices(content);
  content = remapIds(content);
  file.writeAsStringSync(content);
  print('Remapped user references in $path');
}

void main() {
  for (final file in files) {
    processFile(file);
  }
}
